Channel-User: truncate values below threshold in log_plots

kaushik-rohit opened the pull request:
As observed after certain point the log_plot
is insignificant and hence the linear fit
we get is not useful. To fix this, the commit
truncates the degree analysis on CU matrix using
a certain cut-off of 10.
Issue Link: https://github.com/prasadtalasila/IRCLogParser/issues/153

## What? Why?
Fix #153 

Changes proposed in this pull request:
- truncate degree analysis on CU.
- cutoff 10%

## Checks
- [x] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Make sure you have added required documentation
- [x] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Any images?

## Notify reviewers
@rohangoel96

prasadtalasila commented:
@kaushik-rohit The 10% cutoff may be too high for that graph. In the CU graph (figure - 5b in the paper), it is better to truncate when log(Nd) is 0.5. Please translate this heuristic into the corresponding line in ubuntu.py. 

From my understanding of the code, it looks like you are truncating the x-axis at 10. My suggestion is based on y-axis. If you still think your choice is good, we can discuss further on this issue.

It may help both of us decide by looking at the curve fit image (CU_degree_curve_fit.png). The x-axis limit is at around 3.8 and y-axis limit is around 0.5.

kaushik-rohit commented:
@prasadtalasila Isn't both estimation similar (just with a different cutt off to fit the data correctly). When we say cutoff 10 (it should be smaller) we mean cumulative Y is 90% of total sum. Y - log(Nd).  Ignoring y < 0.5 or ignoring values x > x0 where x0 is the point after which y < 0.5 will work atleast in this scenario where we see the graph is decreasing.

prasadtalasila commented:
agreed. As long as 10% cutoff corresponds to 0.5 of log(Nd) in the actual graph, we are fine with the present number.

kaushik-rohit commented:
![cu_degre_curve_fit](https://user-images.githubusercontent.com/22174615/30973119-da26b54c-a45b-11e7-8995-b342653c336f.png)
@prasadtalasila With 0.1 as cutoff this is what the output is. I guess making it 0.5 should give us a much better fit.

kaushik-rohit committed:
Channel-User: truncate values below threshold in log_plots

As observed after certain point the log_plot
is insignificant and hence the linear fit
we get is not useful. To fix this, the commit
truncates the degree analysis on CU matrix using
a certain cut-off of 10.
Issue Link: #153
